snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -280,8 +280,14 @@ def get_dirty_report(self, step):
             differing_options = state.diff_project_options_of_interest(
                 self._project_options)
 
-        if differing_properties or differing_options:
-            return DirtyReport(differing_properties, differing_options)
+        with contextlib.suppress(AttributeError):
```

Comment:
can you add a note as to why this needs to be suppressed, as in the error scenario you are trying to avoid. I have my guesses, and I bet you know today, but I don't want to be scratching my head in the future :-)

Reply:
I've reworked this to check the state before trying to use it. This may be a case where clarity wins over pythonic-ness.